includes/internal/student-progress/lesson-progress/repositories/class-table-reading-aggregate-lesson-progress-repository.php

```diff
@@ -96,11 +96,33 @@ public function save( Lesson_Progress_Interface $lesson_progress ): void {
 				$lesson_progress->get_user_id()
 			);
 		}
+
+		// If the status of the lesson progress is different from the status of the comments based lesson progress,
+		// update the comments based lesson progress to match the status of the lesson progress.
+		// We can't just use the status of the lesson progress because the comments based lesson lesson_progress
+		// has a different underlying set of statuses.
+		if ( $lesson_progress->get_status() !== $comments_based_progress->get_status() ) {
+			if ( ! $comments_based_progress->is_complete() ) {
```

Comment:
I'm not sure I'm getting it, but following the comment (`update the comments based lesson progress to match the status of the lesson progress`), this should be checking the table-based progress, correct?

```suggestion
			if ( $lesson_progress->is_complete() ) {
```